File: apps/api-gateway/internal/models/events.go
package models

import "time"

// Events published to Azure Event Hubs.

type ExamUploadedEvent struct {
	EventID       string    `json:"eventId"`
	EventType     string    `json:"eventType"`
	TenantID      string    `json:"tenantId"`
	Timestamp     time.Time `json:"timestamp"`
	CorrelationID string    `json:"correlationId"`
	ExamID        string    `json:"examId"`
	UploadBatchID string    `json:"uploadBatchId"`
	SourceURI     string    `json:"sourceUri"`
	TotalFiles    int       `json:"totalFiles"`
}

type GradingCompletedEvent struct {
	EventID           string    `json:"eventId"`
	EventType         string    `json:"eventType"`
	TenantID          string    `json:"tenantId"`
	Timestamp         time.Time `json:"timestamp"`
	CorrelationID     string    `json:"correlationId"`
	ExamID            string    `json:"examId"`
	SubmissionID      string    `json:"submissionId"`
	TotalScore        float64   `json:"totalScore"`
	MaxScore          float64   `json:"maxScore"`
	OverallConfidence float64   `json:"overallConfidence"`
}

File: packages/shared-types/src/events.ts
// ─── Event Types (Event Hub Messages) ────────────────────────────────────────

export interface BaseEvent {
  eventId: string;
  eventType: string;
  tenantId: string;
  timestamp: string;
  correlationId: string;
}

export interface ExamUploadedEvent extends BaseEvent {
  eventType: 'ExamUploaded';
  examId: string;
  uploadBatchId: string;
  /** Data Lake URI to the raw upload */
  sourceUri: string;
  totalFiles: number;
}

export interface PageProcessedEvent extends BaseEvent {
  eventType: 'PageProcessed';
  examId: string;
  submissionId: string;
  pageNumber: number;
  imageUri: string;
}

export interface GradingRequestedEvent extends BaseEvent {
  eventType: 'GradingRequested';
  examId: string;
  submissionId: string;
  questionId: string;
  cropUri: string;
  processor: string;
}

export interface GradingCompletedEvent extends BaseEvent {
  eventType: 'GradingCompleted';
  examId: string;
  submissionId: string;
  totalScore: number;
  maxScore: number;
  overallConfidence: number;
}

export type MarkFlowEvent =
  | ExamUploadedEvent
  | PageProcessedEvent
  | GradingRequestedEvent
  | GradingCompletedEvent;
